packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -385,91 +404,553 @@ class ListTile extends StatelessWidget {
     final ThemeData theme = Theme.of(context);
     final ListTileTheme tileTheme = ListTileTheme.of(context);
 
-    final bool isTwoLine = !isThreeLine && subtitle != null;
-    final bool isOneLine = !isThreeLine && !isTwoLine;
-    double tileHeight;
-    if (isOneLine)
-      tileHeight = _denseLayout(tileTheme) ? 48.0 : 56.0;
-    else if (isTwoLine)
-      tileHeight = _denseLayout(tileTheme) ? 60.0 : 72.0;
-    else
-      tileHeight = _denseLayout(tileTheme) ? 76.0 : 88.0;
-
-    // Overall, the list tile is a Row() with these children.
-    final List<Widget> children = <Widget>[];
-
     IconThemeData iconThemeData;
     if (leading != null || trailing != null)
       iconThemeData = new IconThemeData(color: _iconColor(theme, tileTheme));
 
+    Widget leadingIcon;
     if (leading != null) {
-      children.add(IconTheme.merge(
+      leadingIcon = IconTheme.merge(
         data: iconThemeData,
-        child: new Container(
-          margin: const EdgeInsetsDirectional.only(end: 16.0),
-          width: 40.0,
-          alignment: AlignmentDirectional.centerStart,
-          child: leading,
-        ),
-      ));
+        child: leading,
+      );
     }
 
-    final Widget primaryLine = new AnimatedDefaultTextStyle(
+    final Widget titleText = new AnimatedDefaultTextStyle(
       style: _titleTextStyle(theme, tileTheme),
       duration: kThemeChangeDuration,
-      child: title ?? new Container()
+      child: title ?? const SizedBox()
     );
-    Widget center = primaryLine;
-    if (subtitle != null && (isTwoLine || isThreeLine)) {
-      center = new Column(
-        mainAxisSize: MainAxisSize.min,
-        crossAxisAlignment: CrossAxisAlignment.start,
-        children: <Widget>[
-          primaryLine,
-          new AnimatedDefaultTextStyle(
-            style: _subtitleTextStyle(theme, tileTheme),
-            duration: kThemeChangeDuration,
-            child: subtitle,
-          ),
-        ],
+
+    Widget subtitleText;
+    if (subtitle != null) {
+      subtitleText = new AnimatedDefaultTextStyle(
+        style: _subtitleTextStyle(theme, tileTheme),
+        duration: kThemeChangeDuration,
+        child: subtitle,
       );
     }
-    children.add(new Expanded(
-      child: center,
-    ));
 
+    Widget trailingIcon;
     if (trailing != null) {
-      children.add(IconTheme.merge(
+      trailingIcon = IconTheme.merge(
         data: iconThemeData,
-        child: new Container(
-          margin: const EdgeInsetsDirectional.only(start: 16.0),
-          alignment: AlignmentDirectional.centerEnd,
-          child: trailing,
-        ),
-      ));
+        child: trailing,
+      );
     }
 
+    final TextDirection textDirection = Directionality.of(context);
+    final EdgeInsets resolvedContentPadding = contentPadding?.resolve(textDirection)
+      ?? tileTheme?.contentPadding?.resolve(textDirection)
+      ?? const EdgeInsets.symmetric(horizontal: 16.0);
```

Comment:
Maybe convert this magic number into a named constant?

Reply:
OK